recipes/imgui/all/patches/imgui_user_config.patch

```diff
@@ -0,0 +1,12 @@
+--- /dev/null
++++ b/imgui_user_config.h
+@@ -0,0 +1,9 @@
++#pragma once
++
++#ifndef IMGUI_API
++#if defined(_WIN32) && defined(_IMGUI_SHARED)
++#define IMGUI_API __declspec(dllexport)
++#elif defined(_WIN32) && !defined(_IMGUI_STATIC)
++#define IMGUI_API __declspec(dllimport)
++#endif
++#endif
```

Comment:
this is quite confusing.

In this file I would write:

```c++
#ifndef IMGUI_API
#  ifdef IMGUI_SHARED
#    ifdef _WIN32
#      ifdef imgui_EXPORTS
#        define IMGUI_API __declspec(dllexport)
#      else
#        define IMGUI_API __declspec(dllimport)
#      endif
#    else
#      define IMGUI_API __attribute__ ((visibility ("default")
#    endif
#  else
#    define IMGUI_API
#  endif
#endif
```

In CMakeLists:
- define `IMGUI_SHARED` when `BUILD_SHARED_LIBS` is TRUE
- hide all symbols for gnu compilers

In `package_info()`:
- define `IMGUI_SHARED` if shared

Reply:
Did you try to use `generate_export_header` instead? It would avoid patches and interface definitions.